charts/generic-service/README.md

```diff
@@ -67,26 +66,54 @@ hocs-backend-service:
     docsService: https://hocs-docs.{{ .Release.Namespace }}.svc.cluster.local
 ```
 
-### Injecting env into batch yamls
+### overwriting main deployment environment variables with _envs.tpl
+
+You need to set the environment variables used by the main container by creating a file `_envs.tpl`
 
-You can inject the set of environment variables defined for the pods into other application yamls such as batch jobs:
+so for a service with `nameOverride: hocs-templates` you need to do
+
+```yaml
+{{- define "hocs-templates.envs" }}
+- name: JAVA_OPTS
+  value: '{{ tpl .Values.app.env.javaOpts . }}'
+...
+```
+
+This reference in the `define` line is based on the nameOverride field via a helper definition in deployment.yaml.
 
 ```yaml
           env:
-            {{ include "deployment.envs" . | nindent 12 }}
+            {{- include (printf "%s.%s" (include "hocs-deploymentEnvs.name" .) "envs") . | nindent 12 }}
+```
+
+It is possible to override `_envs.tpl` `define` name here without renaming the deployment name by using `deploymentEnvs` 
+
+```yaml
+---
+hocs-generic-service:
+  nameOverride: hocs-templates
+  deploymentEnvs: other-hocs-templates-name
+```
+
+then include a file defined as `other-hocs-templates-name` 
+```yaml
+{{- define "other-hocs-templates-name.envs" }}
+- name: JAVA_OPTS
+  value: '{{ tpl .Values.app.env.javaOpts . }}'
 ```
 
-### Setting environment specific values
+### Setting namespace specific values 
```

Comment:
I appreciate that this is giving us an alternative way of doing it. Why would we choose to do this rather than in `hocs-helmfile`?